src/ceph-volume/ceph_volume/systemd/main.py

```diff
@@ -50,6 +51,29 @@ def parse_osd_uuid(string):
     return osd_uuid
 
 
+def get_block_volume(string):
+    volumes = Volumes()
+    osd_id = string.split('-', 1)[0]
+    osd_fsid = string.split('-', 1)[1]
+    tags={'ceph.type':'block', 'ceph.osd_id':osd_id, 'ceph.osd_fsid':osd_fsid}
+    block_volume = volumes.get(lv_tags=tags).as_dict()
```

Comment:
You need to ensure that ``block_volume`` is not ``None`` before calling ``as_dict()`` here.

Reply:
I think is potentially just a modification to the test though to account for you've changed. The test does not have the correct mocked data to make this call which is why you're getting a ``None`` value.